lib/command.js

```diff
@@ -1743,14 +1756,13 @@ Expecting one of '${allowedValues.join("', '")}'`);
     if (flag.startsWith('--') && this._showSuggestionAfterError) {
       // Looping to pick up the global options too
       let candidateFlags = [];
-      let command = this;
-      do {
+      for (const command of this._getCommandAndAncestors()) {
```

Comment:
Can this be a `forEach`? (I don't like the `for ( of )`.

Reply:
Could be if not for the early exit that is impossible with `forEach`. I guess this is one of the cases where iterating manually via `.parent` is more appropriate.

Reply:
👍